tfx/examples/cifar10/cifar10_pipeline_native_keras.py

```diff
@@ -0,0 +1,205 @@
+  statistics_gen = StatisticsGen(examples=example_gen.outputs['examples'])
+
+  # Generates schema based on statistics files.
+  schema_gen = SchemaGen(
+      statistics=statistics_gen.outputs['statistics'], infer_feature_shape=True)
+
+  # Performs anomaly detection based on statistics and data schema.
+  example_validator = ExampleValidator(
+      statistics=statistics_gen.outputs['statistics'],
+      schema=schema_gen.outputs['schema'])
+
+  # Performs .ations and feature engineering in training and serving.
+  transform = Transform(
+      examples=example_gen.outputs['examples'],
+      schema=schema_gen.outputs['schema'],
+      module_file=module_file)
+
+  # We take the raw examples as the input to the trainer,
+  # as data augmentation needs to be applied on the fly.
+  # When traning on the whole dataset, use 20000 for train steps,
+  # 156 for val steps
+  def _create_trainer(module_file, instance_name):
+    return Trainer(
+        module_file=module_file,
+        custom_executor_spec=executor_spec.ExecutorClassSpec(GenericExecutor),
+        examples=transform.outputs['transformed_examples'],
+        transform_graph=transform.outputs['transform_graph'],
+        schema=schema_gen.outputs['schema'],
+        train_args=trainer_pb2.TrainArgs(num_steps=72),
+        eval_args=trainer_pb2.EvalArgs(num_steps=3),
+        instance_name=instance_name)
+
+  # Uses user-provided Python function that trains a Keras model.
+  trainer = _create_trainer(module_file, 'cifar10')
+
+  # Get the latest blessed model for model validation.
+  model_resolver = ResolverNode(
+      instance_name='latest_blessed_model_resolver',
+      resolver_class=latest_blessed_model_resolver.LatestBlessedModelResolver,
+      model=Channel(type=Model),
+      model_blessing=Channel(type=ModelBlessing))
+
+  # Uses TFMA to compute an evaluation statistics over features of a model and
+  # performs quality validation of a candidate model.
+  eval_config = tfma.EvalConfig(
+      model_specs=[tfma.ModelSpec(label_key='label_xf', model_type='tf_lite')],
+      slicing_specs=[tfma.SlicingSpec()],
+      metrics_specs=[
+          tfma.MetricsSpec(metrics=[
+              tfma.MetricConfig(
+                  class_name='SparseCategoricalAccuracy',
+                  threshold=tfma.config.MetricThreshold(
+                      value_threshold=tfma.GenericValueThreshold(
+                          lower_bound={'value': 0.3}),
+                      change_threshold=tfma.GenericChangeThreshold(
+                          direction=tfma.MetricDirection.HIGHER_IS_BETTER,
+                          absolute={'value': -1e-3})))
+          ])
+      ])
+
+  # Uses TFMA to compute the evaluation statistics over features of a model.
+  # We evaluate using the materialized examples that are output by Transform because
+  # the operations currently performed within Transform are not compatible with TFLite.
```

Comment:
lets be specific here: I think its the decoding_png that is not compatible right?

Reply:
Hmm it's actually not accurate. The reason we take the materialized examples is:
MLKit requires a model that takes the image tensor as input -> I need to specify the serving signature & serve_fn of the model to take float tensors as input -> Evaluator has to take float tensors as input -> Evaluator has to take materialized examples, as raw examples are strings.

Reply:
How about updating the comment like this:

...because the model we trained takes float tensors as the input argument for its serving function.

Reply:
I think the problem is that the input data has serialized png images. This is different from serialized examples. At the same time, as you are pointing out correctly, ML Kit wants deserialized (float32) tensor image inputs. Lets try to capture this.